src/bulk_sampling.jl

```diff
@@ -0,0 +1,151 @@
+
+using AliasTables
+
+# These are exact because their weights sum to powers of 2.
+const ALIASTABLES = (
+    AliasTable{UInt64}(UInt64.([binomial(BigInt(1),i) for i in 0:1])), 
+    AliasTable{UInt64}(UInt64.([binomial(BigInt(2),i) for i in 0:2])),
+    AliasTable{UInt64}(UInt64.([binomial(BigInt(4),i) for i in 0:4])), 
+    AliasTable{UInt64}(UInt64.([binomial(BigInt(8),i) for i in 0:8])),
+    AliasTable{UInt64}(UInt64.([binomial(BigInt(16),i) for i in 0:16])), 
+    AliasTable{UInt64}(UInt64.([binomial(BigInt(32),i) for i in 0:32])),
+    AliasTable{UInt64}(UInt64.([binomial(BigInt(64),i) for i in 0:64]))
+)
+
+function _rand!(rng::AbstractRNG, samples::AbstractArray, m::Memory{UInt64})
+    n = length(samples)
+    n < 100 && return fill_samples!(rng, m, samples)
+    max_i = _convert(Int, m[2])
+    min_i = 5
+    k = 0
+    @inbounds for j in 10235:10266
+        chunk = m[j]
+        if k == 0 && chunk != 0
+            min_i = (j-10235) << 6 + leading_zeros(chunk) + 4
+        end
+        k += count_ones(chunk)
+    end
+    n < 100*(k^0.72) && return fill_samples!(rng, m, samples)
+    inds = Vector{Int}(undef, k)
+    weights = Vector{BigInt}(undef, k)
+    q = 0
```

Comment:
`k` is a more conventional single letter index name than `q` (once we eliminate `i` and `j`).